lib/sass/tree/visitors/perform.rb

```diff
@@ -346,29 +346,47 @@ def visit_mixindef(node)
 
   # Runs a mixin.
   def visit_mixin(node)
+    mixin_name = nil
     @environment.stack.with_mixin(node.filename, node.line, node.name) do
-      mixin = @environment.mixin(node.name)
-      raise Sass::SyntaxError.new("Undefined mixin '#{node.name}'.") unless mixin
-
-      if node.children.any? && !mixin.has_content
-        raise Sass::SyntaxError.new(%(Mixin "#{node.name}" does not accept a content block.))
-      end
-
       args = node.args.map {|a| a.perform(@environment)}
       keywords = Sass::Util.map_vals(node.keywords) {|v| v.perform(@environment)}
       splat = self.class.perform_splat(node.splat, keywords, node.kwarg_splat, @environment)
 
+      mixin_name = node.name
+      # We let any existing mixin named `mixin` shadow dynamic includes
+      # for backwards compatability even though it will break the hell
+      # out of libraries if anyone does this.
+      mixin = @environment.mixin(mixin_name)
+
+      # The while loop is to allow any number of dynamic includes to be chained.
+      while mixin.nil? && mixin_name == "mixin"
```

Comment:
The fully general thing to do here would be to duplicate (or factor out) a bunch of the logic in `funcall.rb` to provide support for arbitrary Ruby-defined mixins with stack-trace rewriting support and everything. That may be overkill, though, since it doesn't seem like we're likely to add more built-in mixins any time soon. The simpler way to get stack trace information for the intermediate `mixin()` calls would be to track the number of nested calls here, and then in the rescue block below do:

```ruby
e.modify_backtrace(mixin: mixin_name, line: node.line)
e.add_backtrace(line: node.line)
dynamic_calls.times do
  e.modify_backtrace(mixin: "mixin", line: node.line)
  e.add_backtrace(line: node.line)
end
```

(This code isn't tested, it may need some tweaks)